streaming/src/channel.cc

```diff
@@ -205,15 +205,23 @@ struct MockQueueItem {
   std::shared_ptr<uint8_t> data;
 };
 
-struct MockQueue {
+class MockQueue {
+ public:
```

Comment:
Should we move all the mock code the to separate test files instead?

Reply:
MockQueue is very simple, perhaps we can move it outside if it becomes complicated.